### ISSUE TITLE
value.go hung on exporting array

### ISSUE THREAD

**Issue by GuyBrand** [robertkrimen/otto#155]
I have an array of values (actually more of js.map) .
If a value contains nil the function hungs on :

} else if state == 1 && kind != t.Kind() {

I've added :
				} else if t == nil {
					fmt.Println("nil !!!", object, index,value)
} else if state == 1 && kind != t.Kind() {

and now it seldom prints "nil !!!" , the dump of object looks like this :
Id,2,StoreNo,8765,CarrierId,,TotalDelivered
Please note the two commas with nothing between them .

1. after this - the function does not hung if that's the right solution, please consider adding it , so I will not fork for this minor issue .
2. what bothers me is that not ll situations that prints the ,, (i.e reached the t==nil) used to hung the function .
3. this does not happen on older version , happen only after I refeteched the code .

Thanks,
   Guy

**Comment by deoxxa** [comment]
Do you have a small example that reproduces this behaviour?

**Comment by deoxxa** [comment]
I'm going to close this for now - feel free to open a new issue if you can provide a small test case.

**Comment by GuyBrand** [comment]
Hi.

Quite a stressed tinge for me.
I will be able to reconstruct this, only now it's a party of a huge code,
so isolating into a small sample well take tinge.

I will reopen the ticket then.

Thanks,
  Guy
